infoblox/resource_infoblox_ip.go

```diff
@@ -126,6 +127,12 @@ func getNextAvailableIPFromCIDR(client *infoblox.Client, cidr string, excludedAd
 
 	network, err := getNetworks(client, cidr)
 
+	if err != nil {
```

Comment:
Does checking for a valid login session at this level make sense?

Surely it would make more sense to do it at infoblox client creation over here https://github.com/prudhvitella/terraform-provider-infoblox/blob/adabc328f9a811acaa09cd9d4b4da650c9026048/infoblox/config.go#L19

Additionally, why re-write the "authentication required" error message, surely whatever message the server gives is more descriptive and will help with better diagnostics.

Reply:
The API returns a HTML document as a string instead of a useful error message. Haven't had a chance to dig into it yet, but you're right re: checking when the client is instantiated as well. This was just to get something working and get Fabio writing some Go. 😛